packages/assets-controllers/src/AccountTrackerController.ts

```diff
@@ -169,58 +240,80 @@ export class AccountTrackerController extends BaseControllerV1<
     }, this.config.interval);
   }
 
+  /**
+   * Refreshes the balances of the accounts using the networkClientId
+   *
+   * @param networkClientId - The network client ID used to get balances.
+   */
+  async _executePoll(networkClientId: string): Promise<void> {
+    this.refresh(networkClientId);
```

Comment:
We don't need mutexes because we await the execution on each cycle right?

Reply:
I think that's true only for the legacy polling loop. But even then if polling is started, stopped, and started again before one poll is able to complete, the mutex would guard against the second execution.

You bring up a good point that `accountsByChainId` state should be read right before it is updated, and then the mutex shouldn't be needed at all.

Reply:
Addressed here: https://github.com/MetaMask/core/pull/3586/commits/ce097429286a20abf1af6434b3a1e0dfdea6df73

Reply:
Mutex added to `refresh` here: https://github.com/MetaMask/core/pull/3586/commits/5f198ac739e00124ccafea35032c5a284368ae34